src/Maestro/Maestro.Web/Startup.cs

```diff
@@ -52,13 +52,22 @@ static Startup()
             {
                 BuildAssetRegistryContext context = entry.Context as BuildAssetRegistryContext;
                 BuildChannel entity = entry.Entity;
-                Build build = entity.Build;
-                if (build.PublishUsingPipelines && ChannelHasAssociatedReleasePipeline(entity.ChannelId, context))
+                Build build = context.Builds.Where(a => a.Id == entity.BuildId).FirstOrDefault();
+
+                if (build == null)
```

Comment:
Is there any situation where this would happen and a retry would be useful?

Debugging one of these is going to be a pain because it would mean:
1. A build was added to BAR
2. A BuildChannel POST with that build + a channel didn't return failure due to the not finding the build
3. We still can't find the build here, and simply skip running the release pipeline, meaning the buildchannel will never actually be inserted.

Reply:
This will be null if-and-only-if someone is attempting to insert a buildchannel using the add-build-to-channel api and the buildid provided isn't valid. We should use context.Builds.Find() to find the build though.

Reply:
> 3\. We still can't find the build here, and simply skip running the release pipeline, meaning the buildchannel will never actually be inserted.

AFAIU The BuildChannel will be inserted as it's not being "cancelled". I'm not sure if a retry here will help.

Reply:
> This will be null if-and-only-if someone is attempting to insert a buildchannel using the add-build-to-channel api and the buildid provided isn't valid. We should use context.Builds.Find() to find the build though.

The API has validation for the BuildId & ChannelId

Reply:
PTAL